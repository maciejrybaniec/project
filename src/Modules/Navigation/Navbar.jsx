/* @flow */
/**
 * Navigation component.
 * @module Modules/Navigation/Navbar
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import cx from 'classnames';
import { pure } from 'recompose';
import React, { PureComponent } from 'react';
import { Link } from 'react-router';

type PropsType = {};

class Navbar extends PureComponent {
    /**
    * Set properties validation for component.
    */
    props: PropsType
    /**
     * Render component in DOM.
     * @returns {ReactElement}
     */
    render(): React.Element<*> {
        return (
            <nav className="Navbar">
                <div className="Navigation__wrapper Container">
                    <div className="Navbar__company">
                        <img
                            className="Navbar__company-image"
                            src="/img/logo.png"
                        />
                    </div>
                    <NavbarMenu className="Navbar__menu" />
                </div>
            </nav>
        );
    }
}

type NavbarMenuProps = {
    className: string
};

/**
 * Navbar menu component.
 * @constructor
 * @param {NavbarMenuProps} props Component properties.
 * @returns {ReactElement}
 */
export const NavbarMenu = pure((props: NavbarMenuProps): React.Element<*> => {
    const { className } = props;
    return (
        <ul className={cx(className)}>
            <li className={cx(`${className}__item`)}>
                <Link
                    className={cx(`${className}-link`)}
                    to={'/about'}
                >
                    O nas
                </Link>
            </li>
            <li className={cx(`${className}__item`)}>
                <Link
                    className={cx(`${className}-link`)}
                    to={'/conditions'}
                >
                    Regulamin
                </Link>
            </li>
        </ul>
    );
});

 export default Navbar;
