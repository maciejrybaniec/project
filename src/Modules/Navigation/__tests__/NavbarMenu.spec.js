/**
 * PageRender specification.
 * @module Modules/Navigation/__tests__/NavbarMenu
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import React from 'react';
import Renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { NavbarMenu } from '../Navbar';

describe('<NavbarMenu />', () => {

    it('should render <NavbarMenu /> component correctly', () => {
        const tree = Renderer.create( <NavbarMenu /> ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should set proper classNames for component elements', () => {
        const componentClassName = 'Navbar__menu';
        const wrapper = mount(<NavbarMenu className={componentClassName} />);

        expect(wrapper.find(`.${componentClassName}-link`).length > 0).toBeTruthy();
        expect(wrapper.find(`.${componentClassName}__item`).length > 0).toBeTruthy();
    });

});
