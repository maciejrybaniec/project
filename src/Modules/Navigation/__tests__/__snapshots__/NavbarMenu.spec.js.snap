exports[`<NavbarMenu /> should render <NavbarMenu /> component correctly 1`] = `
<ul
  className="">
  <li
    className="undefined__item">
    <a
      className="undefined-link"
      onClick={[Function]}
      style={Object {}}>
      O nas
    </a>
  </li>
  <li
    className="undefined__item">
    <a
      className="undefined-link"
      onClick={[Function]}
      style={Object {}}>
      Regulamin
    </a>
  </li>
</ul>
`;
